File: Scala/src/test/sorts/BubbleSortTest.scala
package sorts

import com.ckeyser.sorts.BubbleSort
import com.ckeyser.util.SortUtil
import org.scalatest.FunSpec
import util.TestUtil

/**
  * Created by Chris Keyser on 9/28/2016.
  */
class BubbleSortTest extends FunSpec {
    it("should correctly sort an array") {
        val arr = TestUtil.RandomArray(TestUtil.ARR_SIZE)

        val result = BubbleSort.sort(arr)

        assert(SortUtil.isSorted(result))
    }
}

File: Java/test/sorts/BubbleSortTest.java
package sorts;

import org.junit.Assert;
import org.junit.Test;
import util.SortUtil;
import util.TestUtil;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class BubbleSortTest {
    @Test
    public void BubbleSort_SortsArray() {
        int arr[] = TestUtil.generateRandomArray(TestUtil.ARR_SIZE);

        int[] sorted = BubbleSort.sort(arr);

        Assert.assertTrue(SortUtil.isSorted(sorted));
    }
}
